src/main/java/io/confluent/idesidecar/restapi/util/OperatingSystemType.java

```diff
@@ -0,0 +1,37 @@
+package io.confluent.idesidecar.restapi.util;
+
+import java.util.Locale;
+
+public enum OperatingSystemType {
+  Windows,
+  MacOS,
+  Linux,
+  Other;
+
+  public interface Properties {
+    String getProperty(String key, String defaultValue);
+  }
+
+  @SuppressWarnings("BooleanExpressionComplexity")
+  public static OperatingSystemType from(Properties system) {
```

Comment:
Can you think of a situation in which we need to determine the OS of a system other than the _current_ one?

Reply:
This `from(...)` method doesn't assume the given properties are system properties; I just called the variable that. So technically this `from(...)` method does not assume it's the current system.

The `SidecarInfo` bean's default constructor does use the `System.getProperty(...)` method to determine the OS type, but for testing purposes there is another constructor that the tests can pass other non-`System` properties and verify the behavior of `SidecarInfo` and `OperatingSystemType`.

Reply:
I was asking because I couldn't find an usage of the `from(...)` method with `Properties` other than the System ones so I thought only having `OperationSystemType::current` is enough. I guess it's fine to have the `from(...)` method if we need it for tests later on.

Reply:
Not directly. Most of the test cases in `SidecarInfoTest` do **not** use `System::getProperty(...)` or `System::getenv(...)`, and instead use a map with `Map::get(...)`. See [here](https://github.com/confluentinc/ide-sidecar/pull/76/files#diff-d0cceb0329defee9693d4c44a3c86bd965b6b11774b3e8657a17b56fd140423dR25-R50) and [here](https://github.com/confluentinc/ide-sidecar/pull/76/files#diff-d0cceb0329defee9693d4c44a3c86bd965b6b11774b3e8657a17b56fd140423dR52-R55).